src/Microsoft.ML.Data/Data/Conversion.cs

```diff
@@ -1585,6 +1585,17 @@ public void Convert(in TX span, ref I4 value)
         {
             value = ParseI4(in span);
         }
+        public bool TryConvert(in TX span, ref I4 value)
+        {
+            TryParseSigned(I4.MaxValue, in span, out long? res);
+            if (res.HasValue)
+            {
+                value = (I4)res.GetValueOrDefault();
```

Comment:
>GetValueOrDefault [](start = 32, length = 17)

Why not just GetValue() since you have already checked it has value?

Reply:
> Why not just GetValue() since you have already checked it has value?

Exactly because we already checked the value.  `GetValueOrDefault()` just does `return value;` whereas `.Value` does something like `if (!_hasValue) throw new Exception(); return _value;`.

Reply:
Thanks for the explanation. My understanding of these routines was the exact opposite of what they really do. Looks good!

Reply:
Hmmm. We may have some code to rewrite a number of places...